src/scrapeLinks.test.js

```diff
@@ -1,47 +1,60 @@
 import scrapeLinks from "./scrapeLinks";
 
-const testMarkdownString = `
-This is a Markdown example with [a link to google](https://www.google.com) and [one with a subdirectory](https://www.google.com/nested/page.html)
+const markdownString = `
+Markdown example with a [link to Google](https://www.google.com), one [with a URL path](https://www.google.com/nested/page.html), and others:
 
-and [another to reddit](www.reddit.com) and [a third to Twitter](facebook.com)
+- One [to reddit](www.reddit.com)
+- A fourth [to Facebook](facebook.com) (incomplete URLs)
+- Finally a few [ref] [links][links] [here][link-here]
 
-as well as some blank lines
+[ref]: https://www.ref.com
+[links]:
+  www.links.in/newline
+[link-here]:
+  /just/a/path
+
+There's also some blank lines, misc. text, and <span>HTML</span> code.
 `;
 
 const plaintextString = `
 This string is plaintext, with links like https://www.google.com and https://www.google.com/nested/page.html
 
-I can scrape "https://reddit.com/r/subreddit" and (https://facebook.com) as well!
+I can scrape "https://reddit.com/r/subreddit" and (https://facebook.com) as well! The new regex can pull www.youtube.com too!?
 
-The new regex can pull www.youtube.com too!? unfortunately, gmail.com is just too vague.
+TODO: Unfortunately, gmail.com is just too vague.
+TODO: Ending in a period won't work well either, e.g. www.something.com.
 `;
 
-const plaintextTestResult = [
+const markdownTestResult = [
   "https://www.google.com",
   "https://www.google.com/nested/page.html",
-  "https://reddit.com/r/subreddit",
-  "https://facebook.com",
-  "www.youtube.com",
+  "www.reddit.com",
+  "facebook.com",
+  "https://www.ref.com",
+  "www.links.in/newline",
+  "/just/a/path",
```

Comment:
Vs.

Reply:
Hmmm BTW `jest` is failing with
```
    - Expected  - 2
    + Received  + 0
...
    -   "www.links.in/newline",
    -   "/just/a/path",
```
But if I remove these 2 lines then it fails with
```
    - Expected  - 0
    + Received  + 2
...
    +   "www.links.in/newline",
    +   "/just/a/path",
```

Reply:
Yelp

Reply:
I'm pretty sure you're looking at two different tests, one that splits the string into newlines and one that doesn't.

I just removed the newline split one and changed the inner workings so we don't get string arrays as content (by joining them before returning as content). That should solve this weirdness and allow us to pick up links that take multiple lines like this.

Reply:
Right I thought it had something to do with the new line even though I put `\n` in my regex. I thought maybe it was separately matching those links as plan links which is why I added the plain link in the md sample (https://github.com/iterative/link-check/pull/6#pullrequestreview-695770454 below).

TBH I'm not 100% what the change to the other files do but the tests are passing so thanks! Do you approve this PR now? 🙂

Reply:
> TBH I'm not 100% what the change to the other files do

All just internal stuff, basically this project has a two ways of gathering links: file and GH diff. The GH diff method used to return an array of strings before since I had assumed no links span multiple lines, but that's clearly not the case- now that parser joins the array into a string before returning it, similarly to the filesystem-based parser.

Reply:
> Do you approve this PR now?

Sorry for the delay! I'm trying this PR out manually by running it on the CLI, and noticed some things:

1. dvc.org's prettier rules are butchering these ref links, forcing them onto the same line
2. when forced with `--no-verify`, these links aren't picked up

I test it out manually by running `yarn build`, going to a local `dvc.org` directory, and adding the MD test string to a random MD file. The links were found when prettier forced them on the same line, but multi-line ones aren't found even after all the changes.

Reply:
Just found a change that sort of fixes it, but for some reason `/just/a/path` is still not found.

Reply:
Patch is in! I can deploy tomorrow when I'm a bit less tired, I don't want to rush and mess it up. However, merging in the source repo won't break anything in prod so I can approve/merge. Thanks @jorgeorpinel!

Reply:
Hey thank you so much @rogermparent ! I see you did some more tweaks to address some of the things you mentioned. Let's see what it does on dvc.org once deployed 😬